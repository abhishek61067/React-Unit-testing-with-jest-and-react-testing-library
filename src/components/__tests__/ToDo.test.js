import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import ToDo from "../ToDo";

// cleanup
afterEach(() => {
  cleanup();
});

// test(arg1,arg2)
test("should render incompleted todo", () => {
  const todo = { id: 0, title: "Make dinner", completed: false };
  render(<ToDo todo={todo} />);
  const toDoElement = screen.getByTestId("todo-0");
  expect(toDoElement).toBeInTheDocument();
  expect(toDoElement).toHaveTextContent("Make dinner");
  expect(toDoElement).not.toContainHTML("strike");
});

test("should render completed todo", () => {
  const todo = { id: 1, title: "Make bed", completed: true };
  render(<ToDo todo={todo} />);
  const toDoElement = screen.getByTestId("todo-1");
  expect(toDoElement).toBeInTheDocument();
  expect(toDoElement).toHaveTextContent("Make bed");
  expect(toDoElement).toContainHTML("strike");
});

// snapshot
test("match snapshot", () => {
  const todo = { id: 1, title: "Make bed", completed: true };
  const tree = renderer.create(<ToDo todo={todo} />).toJSON();
  console.log('tree: ',tree);
  expect(tree).toMatchSnapshot();
});
