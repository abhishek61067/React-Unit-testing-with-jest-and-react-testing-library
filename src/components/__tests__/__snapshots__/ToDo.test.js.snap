// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  data-testid="todo-1"
>
  <strike>
    <h1>
      Make bed
    </h1>
  </strike>
</div>
`;
